install.sh

```diff
@@ -1,6 +1,12 @@
 #!/bin/bash
 set -u
 
+# Check if script is run non-interactively (e.g. CI)
+# If it is run non-interactively we should not prompt for passwords.
+if [[ ! -t 0 ]] || [[ ! -t 1 ]] || [[ $SHLVL == 1 ]] || ! tty -s; then
+  NONINTERACTIVE=1
+fi
```

Comment:
I believe you can detect a non-interactive session simpler via `[[ $- != *i* ]]`

Reply:
Yes, $- will only contain `i` if you ask for `$-` in an interactive shell. But if you invoke a shell script from the terminal (and so still have interactive usage over it) this variable does not have `i` anymore. And the install script is not sourced, but is invoked as a shell script on a terminal. 

Just create a script and put `echo $-` in there and you while see when you `bash file.sh`. (Or execute `bash <(echo 'echo $-')`)

Reply:
I came back here to take it back, but again you were faster :)

By coincidence I saw that in other parts this repo tests for interactivity simple with `-t 0`. https://github.com/Homebrew/install/blob/master/uninstall.sh#L287 . I prefer consistency in either direction, but that's just me.